test/controllers/trails_controller_test.rb

```diff
@@ -98,4 +98,72 @@ class Api::V1::TrailsControllerTest < ActionController::TestCase
     assert assigns(:trails)
     assert_equal 1, data.length
   end
+
+  test "can search trails by state" do
+    Trail.create(name: "howdy hike",
+                 city: "Denver",
+                 state: "CO",
+                 country: "United States",
+                 lat: 10.323,
+                 lng: -103.23,
+                 description: "A darn good time",
+                 directions: "take a right")
+    Trail.create(name: "sunny hike",
+                 city: "Denver",
+                 state: "CO",
+                 country: "United States",
+                 lat: 10.323,
+                 lng: -103.23,
+                 description: "A darn good time",
+                 directions: "take a right")
+    Trail.create(name: "rainy bike",
+                 city: "Torrence",
+                 state: "CA",
+                 country: "United States",
+                 lat: 10.323,
+                 lng: -103.23,
+                 description: "A darn good time",
+                 directions: "take a right")
+
+    get :search, state: 'CO', format: :json
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.